specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2019-08-01-preview/Clusters.json

```diff
@@ -0,0 +1,458 @@
+            "required": true,
+            "type": "string",
+            "description": "The resource group name of the cluster."
+          },
+          {
+            "name": "clusterName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Name of the Log Analytics Cluster."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition."
+          },
+          "204": {
+            "description": "NoContent response definition."
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "Clusters"
+        ],
+        "x-ms-examples": {
+          "ClustersGet": {
+            "$ref": "./examples/ClustersGet.json"
+          }
+        },
+        "operationId": "Clusters_Get",
+        "description": "Gets a cluster instance.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The resource group name of the cluster."
+          },
+          {
+            "name": "clusterName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Name of the Log Analytics Cluster."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition.",
+            "schema": {
+              "$ref": "#/definitions/Cluster"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Clusters"
+        ],
+        "x-ms-examples": {
+          "ClustersPatch": {
+            "$ref": "./examples/ClustersUpdate.json"
+          }
+        },
+        "operationId": "Clusters_Update",
+        "description": "Updates a cluster.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The resource group name of the cluster."
+          },
+          {
+            "name": "clusterName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[A-Za-z0-9][A-Za-z0-9-]+[A-Za-z0-9]$",
+            "minLength": 4,
+            "maxLength": 63,
+            "description": "The name of the cluster."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Cluster"
```

Comment:
Patch shouldn't accept the entire resource, only patchable properties should be in the body (no readonly properties).

Please add a new type definition for 'ClusterUpdate' that only contains required fields.

Reply:
Replaced by new type